### DISCUSSION TITLE
Issue with google earth engine

### DISCUSSION THREAD

**Discussion by ZoloKiala** [tethysplatform/tethys#650] [Q&A] [answered]
Dear Tethys developers,

I have been following the tutorial on creating a google earth engine (GEE) app with tethys. I'm stuck with authenticating my app to GEE. When I run the "tethys manage start" command, it says "Not signed up for Earth Engine. Visit https://earthengine.google.com/signup/" I have got an account with GEE and I closely followed the tethys tutorial on GEE. I can't see where I'm getting it wrong.

Regards

Serge

**Comment by swainn** [comment]
Hi Zolo, Sorry for the delay. This sounds more like an issue with Google Earth Engine than an issue with Tethys, but let's get a little more information before we jump to conclusions. Here are a few questions:

Are the native Google Earth engine tools working for you? Here are a few things to try:

- [x] Login to the Code Editor and try running one of the examples: https://code.earthengine.google.com/
- [x] Run through the Google Earth Engine Colab Setup tutorial: https://developers.google.com/earth-engine/guides/python_install-colab  Note: you may need a plugin to use Colab notebooks.

If those don't work for you, then you should reach out to Google Earth Engine to get help with your account: https://developers.google.com/earth-engine/help

If the steps above do work, the provide me with answers to the following questions so I can try to determine if it is an issue with Tethys:

- [x] Are you using the ["stable" version](http://docs.tethysplatform.org/en/stable/tutorials/google_earth_engine.html) of the tutorial? You should see "v: stable" in the header at the top right-hand side of the screen.
- [x] Which step of the tutorial are you on when you encounter the problem?
- [x] Is Google Earth Engine installed in your tethys conda environment? If so what version is installed? Run the following to check:

```bash
conda activate tethys
conda list earthengine-api
```

- [x] Have you authenticated and saved credentials on the machine on which you are running Tethys? See [Primer: Google Earth Engine for Tethys Developers - Step 6](http://docs.tethysplatform.org/en/stable/tutorials/google_earth_engine/part_1/gee_primer.html#authenticate-with-google-earth-engine).
- [x] Are the credentials saved in the ``~/.config/earthengine/`` directory?

```
ls ~/.config/earthengine/
cat ~/.config/earthengine/credentials
```
Note: don't post the contents of the credentials file to public forums like this one.

**Reply by ZoloKiala** [reply]
Dear Nathan,

Thank you very much for your reply. 

Regarding your responses:
1. The native Google Earth engine tools are working fine.
2. I'm using tethys-platform 3.2.2 (I believe it's v stable).
3. I'm the "Visualize Google Earth Engine Datasets" step. I noticed that, when I load a image in app, nothing appears but on the top right side, the attribution "google earth engine" appears.
4. Google Earth Engine is installed in my tethys conda environment. I'm using earthengine-api 0.1.248.
5. The credentials are also saved in the ~/.config/earthengine/ directory

Regards

Serge

**Comment by swainn** [comment]
@ZoloKiala can you share with me the contents of your `install.yml`?

Also, please try updating the `earthengine-api` package to the latest to see if that helps.

```
conda install -c conda-forge earthengine-api=0.1.251
```

**Reply by ZoloKiala** [reply]
Dear Nathan,

Thank you for your reply. See below the content of my install.yml (I removed the comments).  I also did the update.

version: 1.0

name: earth_engine

requirements:

  skip: false
  conda:
    channels:
      - conda-forge
    packages:
      - earthengine-api
      - oauth2client
  pip:

post:
~     

Regards

Serge

**Comment by swainn** [comment]
@ZoloKiala Thank you. Your install.yml looks correct. There was an older version of the tutorial that used a pinned version of earthengine-api that is no longer supported. Yours is not pinned, so that is not the issue.

Here are my next questions/suggestions:

1. Did anything change after updating to the latest version of earthengine-api?
2. Can you share with me the full output of the error you are seeing that appears in your terminal?
3. Try deleting the credentials file from `~/.config/earthengine/` and then run `earthengine authenticate` again.

**Comment by ZoloKiala** [comment]
Thank you, Sir,

 for your reply. I really appreciate it. 

See attached the file with the content of my terminal.

![response_tethys](https://user-images.githubusercontent.com/65705483/108166657-28e40100-70fd-11eb-8d68-d5bc8fdd5c59.PNG)

Regards

Serge

**Comment by swainn** [comment]
@ZoloKiala thank you. More questions:

1. Did you try deleting the credentials file from ~/.config/earthengine/ and then run `earthengine authenticate` again? Did that change anything?
2. Do you have only the default values for `service_account` and `private_key` in `params.py`? In other words, `params.py` should look like this at this point in the tutorial (no credentials defined):

```python
service_account = ''  # your google service account
private_key = ''  # path to the json private key for the service account
```

**Comment by ZoloKiala** [comment, marked as answer]
@swainn thank you again for your help. I deleted the credentials file and run **earthengine authenticate**. Also, I used the default values for **service_account** and **private_key**. It's working now! I'm not getting an error in the terminal. However, the app is no longer loaded in the app library. I restarted supervisorctl, but it was unsuccessful.

**Comment by swainn** [comment]
@ZoloKiala is this a production installation of Tethys Portal? supervisorctl is only used in production. To restart the development server, just press `CTRL-C` in the terminal where it is running and then run `tethys manage start` again.

The most common reason for an app not loading is due to an import error somewhere in the app. To track these issues down I use the following strategy:

1. Activate your Tethys environment:

```
conda activate tethys
```

2. Start-up a Django Console. For this you will need to path to your manage.py, which is located in your Conda environment site-packages/tethys_portal/manage.py:

```
python /path/to/tethys_portal/manage.py shell
```

3. Start importing the module from your app one at a time to see if you run into an import error:

```
>>> from tethysapp.my_app import controllers
>>> from tethysapp.my_app import app
```

**Reply by ZoloKiala** [reply]
@swainn. Thank you again for your assistance. I followed your suggestions. The earth engine app is now loaded. One last issue that persists is that the static files are not loaded properly. I ran "Tethys manage collectall", but it was unsuccessful. See attached the content of the page.

Regards

Serge


![Capture1](https://user-images.githubusercontent.com/65705483/108713053-5a573500-7520-11eb-8673-3bdb0aba9c8e.PNG)

**Comment by swainn** [comment]
@ZoloKiala glad to hear that the app is loaded now. Can you be more specific about `tethys manage collectall` not working? Please share any errors that were printed to the terminal.

**Reply by ZoloKiala** [reply]
@swainn I mean I tried to fix the above error with "tethys manage collectall", but it didn't work. The issue is that the page above does not load my html and css properly. The link of the earth engine app in the app library doesn't work as well.

Regards

Serge

**Reply by swainn** [reply]
@ZoloKiala are you following the instructions in the [Production Installation Guide](http://docs.tethysplatform.org/en/stable/installation/production.html)? This is the section that discusses static files: http://docs.tethysplatform.org/en/stable/installation/production/configuration/basic/static_and_workspaces.html

**Reply by swainn** [reply]
@ZoloKiala perhaps I should clarify first: is this a production installation (using daphne with supervisor to run?) or are you using a development installation (using tethys manage start to run)?

**Reply by ZoloKiala** [reply]
@swainn Thank you once again for your patience. It is a production installation. I delete everything in the system and follow the tutorial again on google earth engine. I noticed, when I get to step "Visualize Google Earth Engine Datasets". My app doesn't load anymore. I even copied the solution codes from GitHub, but it was still unsuccessful. 

Regards

Serge

**Reply by swainn** [reply]
@ZoloKiala can you share the output of the following command:

```
conda activate tethys
conda list
tethys settings --get
```

Also, can you put the broken version of the app code on GitHub and share a link so I can take a look?

**Reply by ZoloKiala** [reply]
@swainn Through our conversations, I managed to find out what was wrong with what I was doing. In fact, I was following the Tethys GEE tutorial using a production installation. Now I did a development installation, everything is working fine. I'm so happy. I wish I could share with you the app that I'm trying to develop, once done. Thank you a lot for your assistance and patience.

Regards

Serge
